cisagov/con-pca-web
# feature CPD-833: retired templates still being used can be seen from the subscription page

#  #

## 🗣 Description ##



When gathering the selected templates while viewing a specific subscription, we gather all templates, both retired and not retired. 

In the select templates dialog box, removed templates which were retired are not shown in the available templates box anymore. They will need to be restored to select them again.



See https://github.com/cisagov/con-pca-web/pull/370 for the necessary backend changes for this feature

## 💭 Motivation and context ##


A template should be able to be retired even when it is currently in use, and should be visible from the edit subscription page even if retired. If a retired template is removed however, it should not be able to be added back without first restoring the template.

Most of the logic preventing retired templates from being selected by new or continuing subscriptions already exists. We simply need to remove the failsafe preventing active subscriptions from being retired.




## 🧪 Testing ##


These changes were tested locally with the following procedure:

1. Choose or create a subscription with multiple templates selected. Make note of which templates are selected.
2. Retire one of the templates being used. Check the subscriptions tab from the template, where you should see the name of your subscription.
3. Return to the subscription page. The retired template should still be visible in the template selection table.
4. Click the change templates button. The retired template should still be visible in the Selected table.
5. Remove the retired template. The template should not be visible in the available table, nor should any other retired templates.
6. Save changes in the Select templates window and Save subscription for the subscription. 
7. The subscription should now be gone from the subscriptions tab of the template.
8. Create a copy of the retired template. You should be able to delete the copy.








## ✅ Pre-approval checklist ##






      in code comments.
      to reflect the changes in this PR.

## ✅ Pre-merge checklist ##





## ✅ Post-merge checklist ##
